core/src/getCollateralPriceOracleConfig.ts

```diff
@@ -0,0 +1,177 @@
+    slot: string,
+    valueIndex?: number,
+    valueToWrite?: BigNumber
+) => {
+    const previousValueRaw = await callFunction(contract, functionName);
+    const previousValue = new BigNumber(valueIndex !== undefined ? previousValueRaw[valueIndex] : previousValueRaw);
+    await overwriteValue(address, valueToWrite || previousValue.plus(1), slot);
+    const currentValueRaw = await callFunction(contract, functionName);
+    const currentValue = new BigNumber(valueIndex !== undefined ? currentValueRaw[valueIndex] : currentValueRaw);
+    const isUnchanged = previousValue.eq(currentValue);
+    if (isUnchanged) {
+        throw new Error('Slot was not specified correctly.');
+    }
+};
+
+const validateConfigWithNextPriceIsValid = async (contract: ethers.Contract, address: string) => {
+    const priceValiditySlotValue = await contract.provider.getStorageAt(
+        address,
+        CONFIG_WITH_NEXT_PRICE.currentPriceSlotAddress
+    );
+    if (priceValiditySlotValue[CONFIG_WITH_NEXT_PRICE.slotPriceValueBeginsAtPosition - 1] !== '1') {
+        throw new Error('Failed to discover the price validity boolean position');
+    }
+    await runOverwriteStep(contract, address, 'peek', CONFIG_WITH_NEXT_PRICE.currentPriceSlotAddress, 0);
+    await runOverwriteStep(contract, address, 'peep', CONFIG_WITH_NEXT_PRICE.nextPriceSlotAddress, 0);
+};
+
+const validateConfigWithoutNextPriceIsValid = async (contract: ethers.Contract, address: string) => {
+    const valueAtSlot = await contract.provider.getStorageAt(
+        address,
+        CONFIG_WITHOUT_NEXT_PRICE.currentPriceValiditySlotAndOffset.slot
+    );
+    if (valueAtSlot[CONFIG_WITHOUT_NEXT_PRICE.currentPriceValiditySlotAndOffset.offset] !== '1') {
+        throw new Error('Failed to discover the price validity boolean position');
+    }
+    await runOverwriteStep(
+        contract,
+        address,
+        'peek',
+        CONFIG_WITHOUT_NEXT_PRICE.currentPriceSlotAddress,
+        CONFIG_WITHOUT_NEXT_PRICE.slotPriceValueBeginsAtPosition
+    );
+};
+
+const run = async () => {
+    await keypress('Press enter to start collateral config generation');
+    await resetNetworkAndSetupWallet(undefined, HARDHAT_PRIVATE_KEY);
+    const basicInfo = await promptBasicInformation();
+    const config: CollateralPriceSourceConfig = basicInfo.hasNextPrice
+        ? CONFIG_WITH_NEXT_PRICE
+        : CONFIG_WITHOUT_NEXT_PRICE;
+    const collateralType = await promptCollateralType();
+    const { contract, address } = await getOracleAddressAndContract(collateralType);
+    console.info(`Contract address: ${address}`);
+    if (basicInfo.hasNextPrice) {
+        const whitelistSlot = 5;
+        const whitelistFunction = 'bud';
+        await addToWhitelist(address, whitelistSlot);
+        const isWhitelisted = await callFunction(contract, whitelistFunction, HARDHAT_PUBLIC_KEY);
+        if (isWhitelisted === '0x00') {
+            throw new Error('Failed to whitelist the wallet on the fork');
```

Comment:
Please provide more logging to make clear what's happening (there is no message that whitelisting even started, no message about which config type is selected, etc). 

Would the developer know what to do when faced with such error message?